### DISCUSSION TITLE
[RFC] Workflows

### DISCUSSION THREAD

**Discussion by auniverseaway** [adobe/da-live#202] [General]
What would you like to see in terms of workflow support? At a high level, I think we would use Workfront for these types of needs.

**To spur some ideas:**

1. If this happens, then do XYZ
2. Scheduling content
3. Content approvals
4. Automated emails based on events (sort of a rehash of item 1)
5. Etc.

The more specific and real-world use case you can be, the better.

**Comment by treeves** [comment]
I'm very very on board with this idea, for multiple reasons.  First off, this would massively help AEM transition projects where folks (correctly) are heavy users of AEM workflows.  Workfront is obviously a class-A product for workflow management, so plugging in Workfront (and providing some starter patterns to follow) for things like ontime/offtime, publish approvals, transactional email, etc.  Or even sidecheck monitoring for things that are concerning our current customer, like people pasting images directly into DA that aren't from the DAM.  If there's a pasted-in image, a workflow could scan for such, put it in a queue to review and then it could get pushed ahead if all a-ok.  

I can also see Workfront being a great intermediary for things like translation workflows, where you need an integration layer to do things like
* Persist a translation job as a unit & get approval to send it
* Cost the job and get approval before pushing it to the translation vendor
* Manage outstanding translation jobs before pulling then back in to DA LOC

And lastly,  I think putting Workfront as the first-class-citizen of workflows for DA will make it easier for Solution Architects (and Adobe Sales) to SELL a solution based around Adobe products that all work together.  Making DA sell-able as a clean, integrated part of the solution is sort of important to ensuring it continues to get the love it deserves.

**Comment by davidbenge** [comment]
fusion would work and there is also this new project called kingfisher built on app builder that could be used.

Kingfisher is a headless workflow engine that powers in-product low-code/no-code experiences for citizen developers.

A common orchestration layer for Adobe Products. The ultimate experience we can provide through project Kingfisher is that an end customer can come into an Adobe product, then through product built UI they can build a workflow or preset (as Firefly calls it) to automate their work. They can run their workflows, view execution details, fetch results, and in future distribute this as an application in a marketplace. 

Fusion has all sorts of built in connectors and blueprints but kingfisher you could bake to be what ever you want it to be and surface it inline anyway you want.

**Comment by jfoxx** [comment]
In writing the workflow, I would like to be able to modify the UI based on context.  I'm thinking specifically about approval workflow. so something like:

1. Publish disable, replaced with Kickoff review
2. Route to {defined in workflow or dialog}
3. Visible "approve" "reject" buttons for approver, with comment fields
4. When approved, publish button is re-enabled
